process/block/interceptedBlocks/interceptedBlockHeader.go

```diff
@@ -86,8 +91,47 @@ func (inHdr *InterceptedHeader) CheckValidity() error {
 	return inHdr.hdr.CheckChainID(inHdr.chainID)
 }
 
+func (inHdr *InterceptedHeader) isEpochCorrect() bool {
+	if inHdr.shardCoordinator.SelfId() != sharding.MetachainShardId {
+		return true
+	}
+	if inHdr.epochStartTrigger.EpochStartRound() >= inHdr.epochStartTrigger.EpochFinalityAttestingRound() {
+		return true
+	}
+	if inHdr.hdr.GetEpoch() >= inHdr.epochStartTrigger.Epoch() {
+		return true
+	}
+	if inHdr.hdr.GetRound() <= inHdr.epochStartTrigger.EpochStartRound() {
```

Comment:
This condition is still necessary if you already have the one from line 107?

Reply:
Theoretically it shouldn't be necessary because the epochstartround should be smaller than the finalityAttestingRound + gracePeriod. Maybe @sasurobert can help here.